WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt

```diff
@@ -132,26 +135,42 @@ open class UploadStarter @Inject constructor(
      * This is meant to be used by [checkConnectionAndUpload] only.
      */
     private suspend fun upload(site: SiteModel) = coroutineScope {
-        val posts = async { postStore.getLocalDraftPosts(site) }
+        val posts = async { postStore.getPostsForSite(site) }
         val pages = async { pageStore.getLocalDraftPages(site) }
 
         val postsAndPages = posts.await() + pages.await()
 
+        // TODO Do we need to check whether the post is currently being editted?
```

Comment:
I've been thinking about this but couldn't think of a scenario where it might be a problem to auto-upload/save in the background while the editor is active. At the same time, I think we should trust the editor to do the remote auto-save itself. 

Do you have a scenario in mind about why we might need this? The one I can think of is when the user changes a published post to Draft using the Post Settings. I'm not sure what will happen. 

We can do it in a separate PR. It seems tricky because we also have to make sure that the editor will always set `isEditingPost(id)` to `false` even when it crashes.

Reply:
> It seems tricky because we also have to make sure that the editor will always set isEditingPost(id) to false even when it crashes.

I don't think we'd need to do this. The [current approach in UploadService](https://github.com/wordpress-mobile/WordPress-Android/blob/3ffcb23f7ea133357b5a8127e616bc1d4f0af29d/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java#L1017) relies on a sticky event which is automatically cleared when the app crashes. 

>Do you have a scenario in mind about why we might need this? The one I can think of is when the user changes a published post to Draft using the Post Settings. I'm not sure what will happen.

What I was worried about is that we update the post in the database on successful upload but the editor won't use the most-up-to-date version. However, I'm not sure if it'll cause any troubles.